fix(content): Various issues in Cognizance

warp-core opened the pull request:
**Bugfix:**

## Fix Details
Various things:
- Typos
- Standardised colon usage: removed unnecessary capitals following colons. The general style throughout the game seems to be to not capitalise after a colon except in cases of proper nouns or similar. This file was not entirely consistent on capitalisation following colons, I have changed all the instances of capitals following colons I have found to lowercase.
- ~~Replace deprecated `apply` conversation nodes with `action`.~~ Handled in another PR.
- Remove ~~a couple of dialogs~~ one (I think) dialog that ~~are~~ is never displayed.
- ~~Remove `commodities "void sprite parts"` from various anonymous fleet definitions in missions since it doesn't do anything and give them the default `cargo 3`. If some specific pool of commodities is desired, it can be updated to pull from a provided list.~~ Handled in another PR.
- Removed the `disables` personality from the Remnant escorts in Cognizance 5: given the orders they were given, they should not be trying to only isable the ships. They still have `plunders` so will still plunder any ships that survive the fight.
- Added an `on enter` dialog for Nenia in Cognizance 5 (open to edits). Gave the Korath ships in that mission `target` so they are marked as mission targets.
- Added a reminder to the `Cognizance 16` `on offer` to use a gaslined ship. While 12 did require one, 14 didn't, and given that I was sent to investigate unknown space, I'm not going to do it in a Puffin.
- Changed the `source` of `Cognizance 17` from a filter for `government Remnant` to `Viminal`. It offers immediately upon completion of 16 which happens on Viminal. There is no reason to use a location filter.
- Removed the `illegal`, `stealth`, and `infiltrating` attributes from Cognizance 32 because they don't do anything on this mission, and, frankly, making getting scanned an atrocity is terrible. Have not yet updated the text. @Zitchas may want to do that themselves, but it should be updated, since the current behaviour does not match the text.

## Testing Done
I haven't tested it yet.

## Save File
This save file can be used to verify the bugfix. The bug will occur when using {{insert commit hash / version}}, and will not occur when using this branch's build.
Just check the spaceport to get the first Cognizance mission.
[CognizanceChanges Tester~3015-01-24 Cognizance available.txt](https://github.com/endless-sky/endless-sky/files/9660022/CognizanceChanges.Tester.3015-01-24.Cognizance.available.txt)

mOctave reviewed (changes requested):
I went through the changes, and found a couple of things. Sorry if the formatting's bad: I haven't done any code review before.

There were a number of places where you removed the capitalization after a colon: To my eye, the capitalization more natural (and is still grammatically correct in American English), than: this. This might not be something to change, especially if the rest of ES follows the British capitalization-after-colon rule.

EjoThims commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -223,13 +223,13 @@ mission "Remnant: Cognizance 5"
 			branch fleetkilled
 				has "remnant: cognizance: killed korath fleet"
-			`When you arrive in the spaceport, you find Prefect Chilia in the middle of briefing a group of Remnant and waves you forward as soon as you approach. "Everyone, this is Captain <last>, who has located the Korath hunters twice now and will take the lead on tracking the rest for this mission." As you cross the room to join him, you take a good look at the group. One detail that stands out to yo ...
+			`When you arrive in the spaceport, you find Prefect Chilia in the middle of briefing a group of Remnant and he waves you forward as soon as you approach. "Everyone, this is Captain <last>, who has located the Korath hunters twice now and will take the lead on tracking the rest for this mission." As you cross the room to join him, you take a good look at the group. One detail that stands out to ...
```
```suggestion
			`When you arrive in the spaceport, you find Prefect Chilia in the middle of briefing a group of Remnant. He waves you forward as soon as you approach and introduces you to the briefing. "Everyone, this is Captain <last>, who has located the Korath hunters twice now and will take the lead on tracking the rest for this mission." As you cross the room to join him, you take a good look at the group. One detail that stands out to you is that every one of the Remnant in this group is wearing a thick leather wrap around their necks, which extends up to their hairline. The leather has a scaly texture, and it appears to be pulsing faintly. Unfortunately, you do not have time to ask questions, as Chilia has not stopped issuing instructions.`
```

Comma absolutely required for the original fix, but this one is a bit less awkward still, as well as smoothing out the other part I found grating.

Zitchas replied:
I think this is an improvement.

EjoThims commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -223,13 +223,13 @@ mission "Remnant: Cognizance 5"
 				goto next
 			label fleetkilled
-			`When you arrive in the spaceport, you find Prefect Chilia in the middle of briefing a group of Remnant and waves you forward as soon as you approach. "Everyone, this is Captain <last>, who has located the Korath hunters twice now and recently destroyed part of their fleet. The Captain will take the lead on tracking the rest for this mission." As you cross the room to join him, you take a good ...
+			`When you arrive in the spaceport, you find Prefect Chilia in the middle of briefing a group of Remnant and he waves you forward as soon as you approach. "Everyone, this is Captain <last>, who has located the Korath hunters twice now and recently destroyed part of their fleet. The Captain will take the lead on tracking the rest for this mission." As you cross the room to join him, you take a g ...
```
```suggestion
			`When you arrive in the spaceport, you find Prefect Chilia in the middle of briefing a group of Remnant. He waves you forward as soon as you approach and introduces you to the briefing. "Everyone, this is Captain <last>, who has located the Korath hunters twice now and recently destroyed part of their fleet. The Captain will take the lead on tracking the rest for this mission." As you cross the room to join him, you take a good look at the group. One detail that stands out to you is that every one of the Remnant in this group is wearing a thick leather wrap around their necks, which extends up to their hairline. The leather has a scaly texture, and it appears to be pulsing faintly. Unfortunately, you do not have time to ask questions, as Chilia has not stopped issuing instructions.`
```

Zitchas replied:
This seems good.

EjoThims commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -223,13 +223,13 @@ mission "Remnant: Cognizance 5"
-			`	He turns to you. "These are the captains who will be following your lead. They are flying two Albatrosses, two Ibises with full complements, and six Starlings. One of the Albatrosses is outfitted to be a shield to take the brunt of attacks." He pauses, looking around at the group. "This task force would be overkill for the intended target, but we are not taking any chances: Make sure every K ...
-			`	He steps back and looks at the group. "May the Embers burn bright for you." With that, he gives the group a respectful nod and walks away. The captains all look around, make various signs you think are probably encouraging combat pilot slang, and variations on Prefect Chilia's parting words. As they disperse to their ships, one gestures to you. "Lead on, Captain."`
+			`	He turns to you. "These are the captains who will be following your lead. They are flying two Albatrosses, two Ibises with full complements, and six Starlings. One of the Albatrosses is outfitted to be a shield to take the brunt of the attacks." He pauses, looking around at the group. "This task force would be overkill for the intended target, but we are not taking any chances: make sure eve ...
+			`	He steps back and looks at the group. "May the Embers burn bright for you." With that, he gives the group a respectful nod and walks away. The captains all look around, and make various signs you think are probably encouraging combat pilot slang and variations on Prefect Chilia's parting words. As they disperse to their ships, one gestures to you. "Lead on, Captain."`
```
```suggestion
			`	He steps back and looks at the group. "May the Embers burn bright for you." With that, he gives the group a respectful nod and walks away. The captains all look around and make various signs you think are probably encouraging combat pilot slang and variations on Prefect Chilia's parting words. As they disperse to their ships, one gestures to you. "Lead on, Captain."`
```

EjoThims commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -223,13 +223,13 @@ mission "Remnant: Cognizance 5"
 			label next
-			`	He turns to you. "These are the captains who will be following your lead. They are flying two Albatrosses, two Ibises with full complements, and six Starlings. One of the Albatrosses is outfitted to be a shield to take the brunt of attacks." He pauses, looking around at the group. "This task force would be overkill for the intended target, but we are not taking any chances: Make sure every K ...
-			`	He steps back and looks at the group. "May the Embers burn bright for you." With that, he gives the group a respectful nod and walks away. The captains all look around, make various signs you think are probably encouraging combat pilot slang, and variations on Prefect Chilia's parting words. As they disperse to their ships, one gestures to you. "Lead on, Captain."`
+			`	He turns to you. "These are the captains who will be following your lead. They are flying two Albatrosses, two Ibises with full complements, and six Starlings. One of the Albatrosses is outfitted to be a shield to take the brunt of the attacks." He pauses, looking around at the group. "This task force would be overkill for the intended target, but we are not taking any chances: make sure eve ...
```
```suggestion
			`	He turns to you. "These are the captains who will be following your lead. They are flying two Albatrosses, two Ibises with full complements, and six Starlings. One of the Albatrosses is outfitted to act as a shield; let it take the brunt of attacks." He pauses, looking around at the group. "This task force would be overkill for the intended target, but we are not taking any chances: make sure every Korath there is killed. Salvage is secondary: priority is making sure there are no escapees."`
```

EjoThims commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -515,7 +515,7 @@ mission "Remnant: Cognizance 14"
 	on stopover
 		conversation
-			`Surprisingly, the station is still operational, despite its apparent age. Ancient environmental systems maintain a breathable atmosphere, although it is thinner than you find comfortable. Likewise, the air is cool - but not dangerously so. Everywhere you look you continually feel like you have stumbled upon a space station built by giants: doors are massive, controls are placed at head height ...
+			`Surprisingly, the station is still operational, despite its apparent age. Ancient environmental systems maintain a breathable atmosphere, although it is thinner than you find comfortable. Likewise, the air is cool - but not dangerously so. Everywhere you look, you continually feel like you have stumbled upon a space station built by giants: doors are massive, controls are placed at head heigh ...
```
Would require adding articles to all of them, and that's a very different tone imo, as it implies a singular set instead of ubiquitous features.

mOctave replied:
Good point.

EjoThims commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -771,7 +770,7 @@ mission "Remnant: Cognizance 21"
 			branch duskknown
 				has "remnant met dusk"
-			`	In the cafeteria the scene hasn't changed much: There is still a large group of Remnant gathered around a cluster of projectors, with a constant stream of gestures discussing various points of the data. You understand enough to figure out that most of the conversation is deep into technical discussions of how the lifesign sensors might give the readings they do without having anything visibl ...
+			`In the cafeteria the scene hasn't changed much: There is still a large group of Remnant gathered around a cluster of projectors, with a constant stream of gestures discussing various points of the data. You understand enough to figure out that most of the conversation is deep into technical discussions of how the lifesign sensors might give the readings they do without having anything visible ...
```
```suggestion
			`In the cafeteria the scene hasn't changed much: there is still a large group of Remnant gathered around a cluster of projectors, with a constant stream of gestures discussing various points of the data. You understand enough to figure out that most of the conversation is deep into technical discussions of how the lifesign sensors might give the readings they do without having anything visible. The principal difference from before is the presence of five pressure-suited Remnant with a neat stack of equipment topped with helmets. One of them notices you and waves you over. "Captain <first>! My name is Dusk, and I am one of the researchers that you volunteered to transport to Ssil Vida."`
```

Looks like you missed one?

EjoThims commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -1330,7 +1325,7 @@ mission "Remnant: Cognizance 30"
 		conversation
 			`Gavriil is waiting at the dock when the <ship>'s hatches finish cycling. "Welcome," they gesture. "It is good to have fresh minds to share <planet> with." The Remnant onboard stream out of the ship in a rush, each carrying a heavy pack of equipment. They gesture respect, and head off in different directions. Evidently they have studied the maps prior to arrival, as they all seem to know where ...
-			`	"We have made good progress since you were here last. We have managed to take over control of life support systems in several wings of rooms, and now those are run entirely from our own systems. Aside from relieving a tiny amount of load from the station's power core, it also means that the station could fail completely and we would still have safe living quarters to serve as base of operati ...
+			`	"We have made good progress since you were here last. We have managed to take over control of life support systems in several wings of rooms, and now those are run entirely from our own systems. Aside from relieving a tiny amount of load from the station's power core, it also means that the station could fail completely and we would still have safe living quarters to serve as a base of opera ...
```
```suggestion
			`	"We have made good progress since you were here last. We have managed to take over control of life support systems in several wings of rooms, and now those are run entirely from our own systems. Aside from relieving a tiny amount of load from the station's power core, it also means that the station could fail completely and we would still have safe living quarters to serve as our base of operations."`
```

Iirc, this was actually still proper as it was, but definitely awkward regardless. This would give a bit more ownership of it though.

EjoThims commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -1604,7 +1597,7 @@ mission "Remnant: Cognizance 38"
 			`	"Fortunately, one of the original abilities of our companions is related to the nervous system. I believe Dawn may have told you a little about it, but was scarce on details." She smiles wryly, then another Remnant takes up the conversation.`
 			`	"To put it simply, we can form ad-hoc neural networks between us, of varying levels of intensity. Without the necessity of speech or gestures, we can have entire conversations in the blink of an eye."`
-			`	He gestures confidently. "A team of Remnant, for instance, could all review incoming data from an experiment, form their own analysis, and then hold what amounts to a conference where each fully presents their thoughts and ideas with supporting data and reasoning within the span of a few minutes."`
+			`	He gestures confidently. "A team of Remnant, for instance, could all review incoming data from an experiment, form their own analyses, and then hold what amounts to a conference where each fully presents their thoughts and ideas with supporting data and reasoning within the span of a few minutes."`
```
This one is a bit tricky.

They are definitely not forming a single common analysis at that point; they each form their own.

But the way it is structured is a bit ambiguous on if it is mentioning the analysis of each individual or the collection of individual analyses.

If the intent is to emphasize them each doing so independently before consulting others, than it is still best to leave it referring to each one's individual analysis.

If the intent is to emphasize the group having the collection to compare, then the change is correct.

Zitchas replied:
The point here is each individual Remnant is forming their own individual idea based on the information. So this is referring to plural analyses.

EjoThims commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -223,13 +223,13 @@ mission "Remnant: Cognizance 5"
 			branch fleetkilled
 				has "remnant: cognizance: killed korath fleet"
-			`When you arrive in the spaceport, you find Prefect Chilia in the middle of briefing a group of Remnant and waves you forward as soon as you approach. "Everyone, this is Captain <last>, who has located the Korath hunters twice now and will take the lead on tracking the rest for this mission." As you cross the room to join him, you take a good look at the group. One detail that stands out to yo ...
+			`When you arrive in the spaceport, you find Prefect Chilia in the middle of briefing a group of Remnant and he waves you forward as soon as you approach. "Everyone, this is Captain <last>, who has located the Korath hunters twice now and will take the lead on tracking the rest for this mission." As you cross the room to join him, you take a good look at the group. One detail that stands out to ...
```
```suggestion
			`When you arrive in the spaceport, you find Prefect Chilia in the middle of briefing a group of Remnant, and he waves you forward as soon as you approach. "Everyone, this is Captain <last>, who has located the Korath hunters twice now and will take the lead on tracking the rest for this mission." As you cross the room to join him, you take a good look at the group. One detail that stands out to you is that every one of the Remnant in this group is wearing a thick leather neck wrap that extends right up to their hairline. It has a scaly texture and appears to be pulsing faintly. Unfortunately, you do not have time to ask questions, as Chilia has not stopped issuing instructions.`
```

warp-core replied:
Is this supposed to be an Oxford comma or is there some other reason to add it?

EjoThims replied:
"He waves you forward..." is a separate independent clause joined with the and conjunction, which requires the comma.

warp-core commented:
I mainly started this to correct some typos/missing words and missions that behaved strangely/badly, then I noticed some other issues while in the files and the colon thing and decided to change that. I wasn't really trying to improve "stilted" lines or poor flow (this would be a lot bigger if I did that). I'm not necessarily opposed to accepting suggestions that are purely related to improving flow and readability, not just grammar, but I'd like @Zitchas to comment on or at least react to those suggestions first.

There are a few other fixes I plan to add today:
Various cargo definitions for fleets are incorrect and don't achieve the desired results.
Some missions that tell me about passengers but don't actually have any `passengers`. Are they sharing my bunk or something?
The missions immediately after taking the void sprites to Esquilline can each offer in any Remnant world. This would be fine, except they lead into each other, including one specifically saying to meet them in the spaceport; we can then go to a different Remnant world and meet them in the spaceport there: that doesn't make sense.
I don't remember the rest right this second.

Some other issues I'd like to raise with @Zitchas beyond what I mentioned in Discord last night:
Instances of the narrator telling me my thoughts and feelings:
When Plume is going for a spacewalk to see the void sprites, the narrator seems to tell me that I wanted to go but have to remind myself that this is Plume's job. First, who says I wanted to go? Second, why can't I go?
Close to that instance, when collecting equipment and crew to take to Gavriil on Esquiline, the narrator tells me what I conclude about the Remnant.
The latter can be resolved fairly easily by just telling me the conclusion, not that it is my conclusion. I would likely just remove the former. I will add direct quotes here shortly.

warp-core commented:
Instances of the narrator telling me what I think or feel:
`Cognizance 1` `on complete`: `Disappointed in the void sprites' behavior, you deliver the recordings to Plume's lab without fanfare.`
`Cognizance 7` `on complete`: `"That's close enough, Captain," says Plume. "I will go outside and watch them from the hull." You help him into a pressure suit and cycle the airlock behind him. You quickly suppress a brief feeling of disappointment. "This is, after all, Plume's job," you say to yourself as you settle back into your chair to wait for him to return.`
`Cognizance 10` `on stopover`: `You shake your head, impressed at how quickly the Remnant make things happen when projects start to move.` (I have changed this in my other changes to that mission.)

Other issues:
`Cognizance 2` `on offer`: Plume tells me to destroy the ship. If I ask if he has the authority to make such an order he says: `I do not have the authority to raise a fleet or order a captain to destroy the ship, but I can declare this Korath a threat to Remnant interests. It is up to you to decide how to respond to the identification of that threat.` Even though he just ordered a captain to destroy the ship. Maybe he exaggerated his request for me to destroy it, or expressed his desire for it to be destroyed inappropriately as an order, but then you should let me call him out on it.
`Cognizance 10` The blocked message doesn't really make sense. What technician? The blocked message also says `his`, but the `on complete` says `she`. I have split this into two missions.
`Cognizance 12` Removed the `cargo 8 equipment` line from the NPC fleet because it doesn't do what is intended. Also made it `accompany save` instead of just `accompany`. I'm not sure which ship name will be used in the `<npc>` replacement, that's something that I'll check if no one else knows already.
`Cognizance 14` What are these long range sensors mentioned in the description?
`Cognizance 14` `on complete` I literally didn't know about the Ember Waste Node until you tell me I noticed it.
`Cognizance 18` It'd be nice if the activities in Nenia took some time and we actually had to fight the Korath ships and defend the Pelicans for a little bit. As it is, all I have to do is enter the system, the Pelicans don't even have to be with me. Similarly, the Pelicans don't have to make it back to Pantica for me to receive the dialog telling me that they've unloaded the Void Sprites into Esquiline. In fact, I can return to Pantica, get that dialog, land on Aventine, and then get nothing because the Pelicans aren't here yet and there is no `on visit`.
`Cognizance 20` The blocked message is pure gameplay: what mission?

tbc

warp-core commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -223,13 +223,13 @@ mission "Remnant: Cognizance 5"
 			label next
-			`	He turns to you. "These are the captains who will be following your lead. They are flying two Albatrosses, two Ibises with full complements, and six Starlings. One of the Albatrosses is outfitted to be a shield to take the brunt of attacks." He pauses, looking around at the group. "This task force would be overkill for the intended target, but we are not taking any chances: Make sure every K ...
-			`	He steps back and looks at the group. "May the Embers burn bright for you." With that, he gives the group a respectful nod and walks away. The captains all look around, make various signs you think are probably encouraging combat pilot slang, and variations on Prefect Chilia's parting words. As they disperse to their ships, one gestures to you. "Lead on, Captain."`
+			`	He turns to you. "These are the captains who will be following your lead. They are flying two Albatrosses, two Ibises with full complements, and six Starlings. One of the Albatrosses is outfitted to be a shield to take the brunt of the attacks." He pauses, looking around at the group. "This task force would be overkill for the intended target, but we are not taking any chances: make sure eve ...
```
The game largely doesn't capitalise after a colon unless it is required.
There are a few news items, one hail from one Person, several instances in the Remnant content written by Zitchas, and a few occurrences in the Deep missions where optional capitalisation is used.
The vast majority of colons are not followed by capitals unless they have to be (including, as far as I can tell, all the colons used by the original developer) and, even in the files I've mentioned where capitalisation is used, it is not consistent and there are cases where the colon isn't followed by a capital.
I think we should be consistent and not capitalising unless it is necessary is the predominant style within the game.

warp-core commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -364,16 +361,16 @@ mission "Remnant: Cognizance 9"
 		conversation
-			`You walk into Plume's lab and are greeted by a large stack of crates and equipment. Plume and another Remnant are gathered around the stack, checking items off a list. "Ah, <first>! Good timing. We were just finishing the inventory of supplies that need to go on this delivery. He gestures to the engineer. "This is Gavriil, one of our best technologists and a skilled astrobiologist in his own  ...
+			`You walk into Plume's lab and are greeted by a large stack of crates and equipment. Plume and another Remnant are gathered around the stack, checking items off a list. "Ah, <first>! Good timing. We were just finishing the inventory of supplies that need to go on this delivery. He gestures to the engineer. "This is Gavriil, one of our best technologists and a skilled astrobiologist in their ow ...
 			`	"It will be a pleasure to work with you," signs Gavriil, making a short bow. "We will be traveling to a research station on <planet>. Yes, it is a gas giant." He smiles at that. "I have to get some equipment set up and get the core systems online. There will be a few more loads of materials afterwards." He looks askance at you. "Shall we get started?"`
```
```suggestion
			`	"It will be a pleasure to work with you," signs Gavriil, making a short bow. "We will be traveling to a research station on <planet>. Yes, it is a gas giant." They smile at that. "I have to get some equipment set up and get the core systems online. There will be a few more loads of materials afterwards." They look askance at you. "Shall we get started?"`
```

Zitchas commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -53,14 +53,13 @@ mission "Remnant: Cognizance 1"
 		fleet
 			names "korath"
-			cargo 1
-			commodities "void sprite parts"
```
Might be worth noting that there *is* a "void sprite parts" commodity list (see line 869 in `commodities.txt`), and my intent was that this pulls one option from that list to fill a random amount of cargo space.

Having a random 3 cargo doesn't have the same impact.

And yes, while having some void sprite parts is just a junk cargo for the moment, these are intended to be used for future side missions.

warp-core replied:
It's not just that it adds "junk cargo." It is totally ignored.
If it were given a price range like that normal trade commodities, it would be possible for it to show up here, but that would also mean it appears in the trade tab on every planet that has one, which is nonsensical.

![image](https://user-images.githubusercontent.com/20605679/191547321-3d6766cf-daf6-453d-8d40-6647de045023.png)

warp-core replied:
I have given them all `cargo 0`, so now they won't have any commodities (and can't have outfits because there are no outfitters near where they spawn).

Zitchas commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -116,23 +114,22 @@ mission "Remnant: Cognizance 2"
 			names "korath"
-			cargo 1
-			commodities "void sprite parts"
+			cargo 3
```
Likewise here: There *is* a "void sprite parts" commodity listed in `commodities.txt`, and this is supposed to pull from that list. If that's not working, I'd like to fix it rather than replace that cargo with random miscellaneous stuff.

Zitchas commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -177,8 +174,7 @@ mission "Remnant: Cognizance 4"
 			names "korath"
-			cargo 2
-			commodities "void sprite parts"
+			cargo 3
```
Likewise, I'd prefer to ensure the intended behavior is happening, rather than replacing it with a non-intended generic behavior.

Zitchas commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -233,20 +229,19 @@ mission "Remnant: Cognizance 5"
 			names "korath"
-			cargo 2
-			commodities "void sprite parts"
+			cargo 3
```
Same as previous instances.

Zitchas commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -394,41 +391,54 @@ mission "Remnant: Cognizance 9"
+
+
+
+mission "Remnant: Cognizance 10B"
+	landing
+	name "Technician and supplies to <planet>"
+	description "Transport this technician and their equipment to <destination>."
+	source "Caelian"
+	destination "Esquiline"
+	blocked "Plume's technician needs a free bunk and <tons> of cargo space for her equipment. Come back when you have enough space."
+	cargo equipment 8
+	passengers 1
+	to offer
+		has "Remnant: Cognizance 10A: done"
+	on accept
 		dialog
-			`Flight control gives you directions to a landing pad that already has a large pile of crates stacked next to it. As you land, workers swarm out to help load the equipment. Once everything is stowed, a young technician climbs aboard, stows their gear, and lets you know that everything is ready for departure. You shake your head, impressed at how quickly the Remnant make things happen when proj ...
+			`Flight control gives you directions to a landing pad that already has a large pile of crates stacked next to it. As you land, workers swarm out to help load the equipment. Once everything is stowed, a young technician climbs aboard, stows their gear, and lets you know that everything is ready for departure. The Remnant can certainly make things happen quickly when projects start to move.`
 	on complete
 		payment 100000
 		conversation
 			`On the flight over, the technician joins you on the control deck to chat. She's a biotech specialist, adept at managing life-support systems in alien atmospheres, and a recent graduate from a Remnant educational program that seems to be similar to a university. Apparently she studied astrobiology and is part of a project aimed at supporting mutually exclusive environments in close proximity.  ...
 			`	As you approach <planet>, it occurs to you that the Remnant might appreciate you taking steps to maintain the secrecy of the station. Altering the <ship>'s path slightly, you swing down into <planet> in an oblique path then meander around the planet under the cloud layer until arriving at the station. Your passenger nods approvingly as she notices the flight path. "Secrecy is always good wit ...
-			`	The <ship> settles into the landing bay, and the technician quickly disembarks. Apparently there must have been other shipments while you were gone, as there are now a handful of people visible and quite a bit more equipment stacked everywhere. The station itself feels more alive, the air fresher. As before, you promptly receive notification of another payment of <payment> to your account.`
+			`	The <ship> settles into the landing bay, and the technician quickly disembarks. Apparently, there must have been other shipments while you were gone, as there are now a handful of people visible and quite a bit more equipment stacked everywhere. The station itself feels more alive, the air fresher. As before, you promptly receive notification of another payment of <payment> to your account.`
 
 
 
 mission "Remnant: Cognizance 11"
+	landing
 	name "Remnant Data Delivery"
 	description "Deliver a data crystal to the engineers on <planet>."
-	landing
 	source "Esquiline"
 	destination "Viminal"
 	to offer
-		has "Remnant: Cognizance 10: done"
+		or
+			has "Remnant: Cognizance 10: done"
+			has "Remnant: Cognizance 10B: done"
```
I'm not entirely sure what is going on here, I'll have to re-read this later. (and this is as far as I can review at the moment).

That being said, should that "or" statement be checking for 10A or 10B instead of 10 or 10B? (since you replaced 10 with 10A above)?

warp-core replied:
> That being said, should that "or" statement be checking for 10A or 10B instead of 10 or 10B? (since you replaced 10 with 10A above)?

I have split 10 into 10A and 10B.
10A is the trip from Esquiline to Caelian, 10B is the trip back to Esquiline, rather than Caelian being a stopover.
11 should require a person to have either done 10 (if they picked up the mission before this change) or 10B.

Zitchas reviewed (changes requested):
Overall looks like some good changes. This isn't a complete review, just as far as I can get right now.

Main thing is probably regarding the "void sprite parts" actually being in the commodity list, and I'd like to keep them being used there. rather than replacing them with generic cargo.

As an aside, that's probably why the game isn't throwing an error message for them: They are, in fact, already listed as a valid commodity list in commodities.txt.

warp-core commented:
> Main thing is probably regarding the "void sprite parts" actually being in the commodity list, and I'd like to keep them being used there. rather than replacing them with generic cargo.
> 
> As an aside, that's probably why the game isn't throwing an error message for them: They are, in fact, already listed as a valid commodity list in commodities.txt. (see line 869 in said commodities file)

The `commodities` part of a cargo definition for a fleet allows the *trade* commodities (the things you can buy and sell in the trade panel) that ships in the fleet might carry to be defined (default is any of them). It does not pull commodities from the commodities.txt file which I believe is used when randomly selecting cargo for jobs. Have you ever encountered a randomly spawned or mission NPC ship with cargo that uses any of the names from the commodities.txt file and not just one of the trade commodities?

You could perhaps create "void sprite parts" outfits (the Remnant might even buy them from you since the Archon discouraged them acquiring them directly themselves, "Prometheus is punished for stealing from the [Void Sprites], but no one is punished for stealing from Prometheus," or however it goes) and have these fleets pull outfits from an outfitter only containing those items.

EjoThims commented:
I'm liking the sound of that idea.

warp-core commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -373,7 +373,7 @@ mission "Remnant: Cognizance 9"
 		log "People" "Gavriil" `A technician that works with Plume, and a talented astrobiologist. Gavriil is the project lead for the repurposed Esquiline station.`
 		conversation
-			`You walk into Plume's lab and are greeted by a large stack of crates and equipment. Plume and another Remnant are gathered around the stack, checking items off a list. "Ah, <first>! Good timing. We were just finishing the inventory of supplies that need to go on this delivery. He gestures to the engineer. "This is Gavriil, one of our best technologists and a skilled astrobiologist in his own  ...
+			`You walk into Plume's lab and are greeted by a large stack of crates and equipment. Plume and another Remnant are gathered around the stack, checking items off a list. "Ah, <first>! Good timing. We were just finishing the inventory of supplies that need to go on this delivery. He gestures to the engineer. "This is Gavriil, one of our best technologists and a skilled astrobiologist in their ow ...
```
> Gabriel is male, but androgenous. Mixing terms was an intentional choice to convey that confusion.

Not sure this is a good way to convey that. It just looks like typos.

warp-core commented:
Oh, something I meant to mention but don't think I did:
Why is the Bellows not added to the outfitter when the Merganser is added to the shipyard?
There might be the argument that they can't make enough to sell openly, but they can make enough to outfit as many Mergansers as I can buy with them. And I can just take them off of those and sell the Mergansers back at full price (minus the cost of the Bellows I've uninstalled) and install them on my own ship. You've just made it inconvenient for me.
Also, why are the Mergansers added to shipyards immediately? Could they instead be added where the Bellows currently is (at the same time that peregrines get added to Remnant fleets)? Otherwise, I think the Bellows should be sold earlier.

Zitchas commented:
Mostly because that's where it felt right to make them available. If the convenience issue is really that big a deal I'd rather move the bellows up, I think.

warp-core commented:
I've added a save file that places you just before the first Cognizance mission if anyone wants to play with these changes.

Zitchas commented:
PR #7281 should fix some of those things you highlighted here. Specifically, the void sprite parts cargo.

Zitchas reviewed (changes requested):
Would it be possible to have you pull the major structural changes in Cognizance 10 (basically, all the stuff involved in splitting the mission in two) into a separate PR please? 

I think I can get through everything else in this PR reviewed in a fairly straight forward manner (few major changes, maybe just outright approving it), but I'm having trouble parsing and dealing with everything going on in that splitting section. Having *just* the stuff required to split the mission in a stand-alone PR would make it much easier to understand and deal with.

I know it's a bunch of extra work, but it'll be very much appreciated, and significantly improves the likelyhood of my being able to finish reviewing any of this in the immediate future.

Zitchas commented:
Thank you. Are you going to be implementing and/or resolving Ejothms' and mOctave's suggestions?

warp-core commented:
> Thank you. Are you going to be implementing and/or resolving Ejothms' and mOctave's suggestions?

Of those comments, the ones which remain unresolved I think are largely literary style changes, not necessarily about correctness or fitting in with the rest of the game. I have no strong opinions on them either way, so I'd prefer if you commented/reacted to any of those which you would like added (since, personally, I don't think I'm inclined to add any of them unless Ejo can elaborate on any that he thinks are required for correctness, with the possible exception of https://github.com/endless-sky/endless-sky/pull/7261#discussion_r974836793.)

Zitchas commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -223,13 +223,13 @@ mission "Remnant: Cognizance 5"
 			branch fleetkilled
 				has "remnant: cognizance: killed korath fleet"
-			`When you arrive in the spaceport, you find Prefect Chilia in the middle of briefing a group of Remnant and waves you forward as soon as you approach. "Everyone, this is Captain <last>, who has located the Korath hunters twice now and will take the lead on tracking the rest for this mission." As you cross the room to join him, you take a good look at the group. One detail that stands out to yo ...
+			`When you arrive in the spaceport, you find Prefect Chilia in the middle of briefing a group of Remnant and he waves you forward as soon as you approach. "Everyone, this is Captain <last>, who has located the Korath hunters twice now and will take the lead on tracking the rest for this mission." As you cross the room to join him, you take a good look at the group. One detail that stands out to ...
```
Looks like a good change.

EjoThims commented:
Generally the smaller the change, the more likely it's just a pure fix.

Sometimes straight fixes are awkward, though, especially when the "incorrect" version "feels" right instead of just being mistakes in typing.

Sometimes the best fix for a simple mistake is an extensive change.

Best example I can think of off the top of my head is sentence fragments incorrectly appended to already long sentences. Just adding a missing comma or conjunction is the simplest correction, but then you run afoul of looser conventions, like not combining more than two independent clauses with simple conjunctions, no matter how many dependent clauses or phrases are also spliced in. It's not -technically- incorrect, even by the strictest standards, afaik, it's just really not done, and there's almost always a better way to split it up.

mOctave reviewed (commented):
This looks good now. I just found one possible grammar error, but even that might be fine as is.

Zitchas commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -219,21 +219,21 @@ mission "Remnant: Cognizance 5"
 			label next
-			`	He turns to you. "These are the captains who will be following your lead. They are flying two Albatrosses, two Ibises with full complements, and six Starlings. One of the Albatrosses is outfitted to be a shield to take the brunt of the attacks." He pauses, looking around at the group. "This task force would be overkill for the intended target, but we are not taking any chances: make sure eve ...
-			`	He steps back and looks at the group. "May the Embers burn bright for you." With that, he gives the group a respectful nod and walks away. The captains all look around, and make various signs you think are probably encouraging combat pilot slang and variations on Prefect Chilia's parting words. As they disperse to their ships, one gestures to you. "Lead on, Captain."`
+			`	He turns to you. "These are the captains who will be following your lead. They are flying two Albatrosses, two Ibises with full complements, and six Starlings. One of the Albatrosses is outfitted to act as a shield; let it take the brunt of attacks." He pauses, looking around at the group. "This task force would be overkill for the intended target, but we are not taking any chances: make sur ...
```
The priority is probably more correct, but this is a spoken statement where the Disney had dropped the "the".

mOctave reviewed (approved):
All my issues with this have been addressed.

warp-core commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -369,17 +370,17 @@ mission "Remnant: Cognizance 9"
-	passengers 1
 	source "Aventine"
 	destination "Esquiline"
+	cargo equipment 8
```
```suggestion
	cargo "equipment" 8
```

warp-core commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -1072,12 +1064,12 @@ mission "Remnant: Cognizance 23"
 		government "Remnant"
 	stopover "Esquiline"
 	destination "Ssil Vida"
+	cargo "Outpost Equipment" 11
```
```suggestion
	cargo "Outpost Equipment" 11
```
```suggestion
	cargo "outpost equipment" 11
```

warp-core commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -1365,13 +1357,9 @@ mission "Remnant: Cognizance 31"
-	infiltrating
 	source "Viminal"
 	destination "Clink"
+	deadline 19
```
```suggestion
	blocked "Chilia and Taely look excited about something, but when you inquire, they shake their heads and say you need to have earned their capital license before they can tell you more."
	deadline 19
```

warp-core reviewed (commented):
Some of the research lab related missions could probably also be improved with the new `to accept` thing, I'll probably do those separately.

Zitchas commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -666,11 +664,11 @@ mission "Remnant: Cognizance 19"
 	description "Deliver navigational data on the new systems to <destination>."
-	source
-		government "Remnant"
+	source "Aventine"
```
```suggestion
	source
	 	 government "Remnant"
	 	 neighbor planet "Aventine" "Caelian"
```

This mission was intentionally setup (and written) with the intention that it could trigger on any Remnant location, including stations, and any other permanent bases that are revealed in future content. A better source filter, for now, would be a Remnant planet that is neighbor to Aventine or Caelian. (with both listed, that should enable everywhere in the core trio of systems.

I'm not 100% that I did the neighbor correctly, but hopefully I did. The wiki says multiple planets can be listed in sequence.

warp-core replied:
If that's what you want, it should probably just be:
```
government "Remnant"
near <any of those three systems> 1
```
But, given that this mission can offer as soon as Cognizance 18 is complete, which occurs on Aventine, there's not much reason to have this mission offer anywhere else.

Zitchas replied:
OK, updated the suggestion above. Thanks!

warp-core replied:
There's still the fact that this mission will *never* offer anywhere but Aventine anyway, unless you specifically exclude Aventine.

Zitchas replied:
So the fact that it could, theoretically, if the player managed to get there somehow, is in this case more of a safety feature in case someone manages to break something or some corner case we aren't aware of results in it not triggering there this one time.

Unless this is actually a problem, I don't see a reason why allowing it to trigger elsewhere (as a safety feature if nothing else) should be an issue.

warp-core replied:
What it does is unnecessarily obfuscate what is actually going on. We shouldn't make stuff needlessly complicated as a safety feature in case something breaks, especially when you cannot even describe the mechanism that would have to break. Even if the breakage you suggest occurred, a person would still be able to get this mission upon returning to Aventine anyway.

Zitchas replied:
All it takes is one mission that for some reason takes priority here to block this from happening. 

As far as that goes, I've had this not trigger the first time. No idea why (hasn't happened recently, so I can't provide circumstances or saves), but in my experience the logic for missions firing (especially since the game doesn't let more than one mission offer in the same place at a time) isn't iron-clad.

That being said "any Remnant planet within 2 jumps of Arculus" is hardly obfuscated or hard to understand.

warp-core replied:
> All it takes is one mission that for some reason takes priority here to block this from happening.

`priority` does not affect landing missions.

The letter thing is well-defined behaviour that people are taking advantage of. Not a vague "if something breaks."

But, actually, I have thought of a reasonable way this could be prevented from offering, though I don't think this arrangement is possible in the vanilla game right now, so sure. I do have other issues with this mission and the following ones which I've outlined in a comment near the top of this, but I don't want to deal with changing that stuff right now.

Zitchas replied:
alright.

Zitchas commented on data/remnant/remnant 2 cognizance.txt:
```diff
@@ -1383,7 +1372,7 @@ mission "Remnant: Cognizance 32"
 				has "Sheragi Archaeology: Epilogue: done"
 			`	A hologram of Ssil Vida appears in the air over the table, then fades away to leave a large highlighted object with a network of lines reaching around the station. "As Gavriil has told you, the station's primary power source appears to be a massive fusion reactor. As best we can tell, it is quite a simple design, albeit one built for totally automated functioning and bigger than the largest  ...
-			`	"Based on our analysis, the reactor appears to use a very specific isotope. Whether it is truly needed or others would suffice is unknown, but for now we are inclined to provide it with an exact match. While we will be able to produce it eventually, it will take time that we do not have." The diagram of Ssil Vida vanishes and is replaced by a map of the galaxy, which focuses in on <system>,  ...
+			`	"Based on our analysis, the reactor appears to use a very specific isotope. Whether it is truly needed or others would suffice is unknown, but for now we are inclined to provide it with an exact match. While we will be able to produce it eventually, it will take time that we do not have." The diagram of Ssil Vida vanishes and is replaced by a map of the galaxy, which focuses in on <system>,  ...
```
```suggestion
			`	"Based on our analysis, the reactor appears to use a very specific isotope. Whether it is truly needed or others would suffice is unknown, but for now we are inclined to provide it with an exact match. While we will be able to produce it eventually, it will take time that we do not have." The diagram of Ssil Vida vanishes and is replaced by a map of the galaxy, which focuses in on <system>, near the juncture of the Rim and the South. "Our records indicate that the most easily accessible deposits are on <planet>." Chilia turns to you. "As you are from ancestral space and are most familiar with the people there, we would like you to travel there and bring back 58 cubic meters of processed fuel for the reactor. Can you do this?"`
```

While the rest of humanity may use tons, the Remnant do not. Given that the word "ton" does, in fact, refer to 1000kg (or 1 cubic meter of water), the reference here to cubic meters does not conflict with the use of ton elesewhere. (Yes, I am well aware that ton *can* refer to about a dozen other potential quantities, but humanity is standardizing on metric, so I strongly suspect that our "ton" is specifically the 1000kg one).

warp-core replied:
> Given that the word "ton" does, in fact, refer to 1000kg (or 1 cubic meter of water), the reference here to cubic meters does not conflict with the use of ton elesewhere.

The player should not be expected to guess at what the meaning of these units is. If you want to use something other than `tons`, it should just be `units`, since that is a generic term.

Zitchas replied:
There is also the fact that this is within dialog. NPCs are allowed to say things that may confuse the player and/or not conform to the rest of the game.

Zitchas reviewed (changes requested):
Thanks for going through this. Definitely quite a few corrections needed. The spelling and grammatical ones all look good, but there's a couple wording changes that I don't agree with.

Zitchas reviewed (approved):
I think this looks good. Thanks for making the list of further issues to work on, too. Those should be helpful in refining these in the future. Particularly the instances of telling the player what they think.

Zitchas reviewed (approved):
Re-approving, just to be thorough.
